fe/fe-core/src/main/java/com/starrocks/planner/NestLoopJoinNode.java

```diff
@@ -2,32 +2,73 @@
 
 package com.starrocks.planner;
 
-import com.google.common.base.Joiner;
 import com.google.common.base.Preconditions;
 import com.starrocks.analysis.Expr;
 import com.starrocks.analysis.JoinOperator;
+import com.starrocks.analysis.SlotRef;
 import com.starrocks.analysis.TableRef;
+import com.starrocks.common.IdGenerator;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.SessionVariable;
 import com.starrocks.thrift.TNestLoopJoinNode;
 import com.starrocks.thrift.TPlanNode;
 import com.starrocks.thrift.TPlanNodeType;
 import org.apache.commons.collections.CollectionUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
 
 /**
  * NESTLOOP JOIN
  *  TODO: support all kinds of join type
  */
-public class NestLoopJoinNode extends JoinNode {
+public class NestLoopJoinNode extends JoinNode implements RuntimeFilterBuildNode {
+
     private static final Logger LOG = LogManager.getLogger(NestLoopJoinNode.class);
 
     public NestLoopJoinNode(PlanNodeId id, PlanNode outer, PlanNode inner, TableRef innerRef,
                             JoinOperator joinOp, List<Expr> eqJoinConjuncts, List<Expr> joinConjuncts) {
         super("NESTLOOP JOIN", id, outer, inner, joinOp, eqJoinConjuncts, joinConjuncts);
+    }
+
+    @Override
+    public void buildRuntimeFilters(IdGenerator<RuntimeFilterId> generator) {
+        SessionVariable sessionVariable = ConnectContext.get().getSessionVariable();
+        JoinNode.DistributionMode distributionMode = JoinNode.DistributionMode.BROADCAST;
```

Comment:
Why it must be BROADCAST?

Reply:
Right table of NestloopJoin could only be gathered or broadcast, so the filter must be broadcasted.

Reply:
At present, does rhs of NLJ always come from broadcast exchange sink?
bucket_shuffle/collocated/replicated JOIN should be also taken into consideration, but in these cases, has_remote_target should be set to false.  These rf take effects across-fragment-instance appears dangerous.

Reply:
NestloopJoin does not support partitioned join, so bucket_shuffle/colocated is not taken into consider. And right table could only be gathered or broadcasted.